simple_history/models.py

```diff
@@ -133,6 +133,28 @@ def finalize(self, sender, **kwargs):
         setattr(sender, self.manager_name, descriptor)
         sender._meta.simple_history_manager_attribute = self.manager_name
 
+    def get_history_model_name(self, model):
+        if not self.custom_model_name:
+            return "Historical{}".format(model._meta.object_name)
+        # Must be trying to use a custom history model name
+        if callable(self.custom_model_name):
+            name = self.custom_model_name(model._meta.object_name)
+        else:
+            #  simple string
+            name = self.custom_model_name
+        # Desired class name cannot be same as the model it is tracking
+        if (
+            name.lower() != model._meta.object_name.lower()
+            or model.__module__ != self.module
```

Comment:
This is trying to cater for models registered from an external app and would therefore have a different module path than the history being created so even if the class names are the same they are not in the same package which would be ok

Reply:
Can change logic to:
`not (name.lower() == model._meta.object_name.lower()
            and model.__module__ == self.module)`

Reply:
I prefer that logic. However, the real issue is that `self.module` might not actually tell you the module. If you look at https://github.com/treyhunner/django-simple-history/blob/master/simple_history/tests/models.py#L405 in tests, `self.module` is actually the module of its inherited Abstract class rather than `tests.models`. I think you need to use `model.__module__ == attrs["__module__"]`, but that should be tested regardless. Ideally you'd have a test showing why the current conditional doesn't work

Reply:
@uhurusurfa

Reply:
Hey @uhurusurfa are you still working on this?